servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/ReplayStrategies.java

```diff
@@ -0,0 +1,371 @@
+     * @param history max number of signals (excluding {@link Subscriber#onComplete()} and
+     * {@link Subscriber#onError(Throwable)}) to retain.
+     * @param <T> The type of {@link ReplayStrategyBuilder}.
+     * @return a {@link ReplayStrategyBuilder} using the history strategy.
+     */
+    public static <T> ReplayStrategyBuilder<T> historyBuilder(int history) {
+        return new ReplayStrategyBuilder<>(() -> new MostRecentReplayAccumulator<>(history));
+    }
+
+    /**
+     * Create a {@link ReplayStrategyBuilder} using the historyHint and TTL strategy.
+     * @param historyHint hint for max number of signals (excluding {@link Subscriber#onComplete()} and
+     * {@link Subscriber#onError(Throwable)}) to retain. Due to concurrency between threads (timer, accumulation,
+     * subscribe) the maximum number of signals delivered to new subscribers may potentially be more but this hint
+     * provides a general bound for memory when concurrency subsides.
+     * @param ttl duration each element will be retained before being removed.
+     * @param executor used to enforce the {@code ttl} argument.
+     * @param <T> The type of {@link ReplayStrategyBuilder}.
+     * @return a {@link ReplayStrategyBuilder} using the historyHint and TTL strategy.
+     */
+    public static <T> ReplayStrategyBuilder<T> historyTtlBuilder(int historyHint, Duration ttl, Executor executor) {
+        return historyTtlBuilder(historyHint, ttl, executor, false);
+    }
+
+    /**
+     * Create a {@link ReplayStrategyBuilder} using the historyHint and TTL strategy.
+     * @param historyHint hint for max number of signals (excluding {@link Subscriber#onComplete()} and
+     * {@link Subscriber#onError(Throwable)}) to retain. Due to concurrency between threads (timer, accumulation,
+     * subscribe) the maximum number of signals delivered to new subscribers may potentially be more but this hint
+     * provides a general bound for memory when concurrency subsides.
+     * @param ttl duration each element will be retained before being removed.
+     * @param executor used to enforce the {@code ttl} argument.
+     * @param lazy
+     * <ul>
+     *     <li>{@code true} will evict expired items in a lazy fashion when new subscribers arrive. This approach
+     *     is more likely to retain {@code historyHint} elements in memory in steady state, but avoids cost of
+     *     scheduling timer tasks.</li>
+     *     <li>{@code false} will evict expired items eagerly when they expire. If {@code ttl} is lower that
+     *     {@code historyHint} relative to signal arrival rate this can use less memory but schedules time tasks.</li>
+     * </ul>
+     * @param <T> The type of {@link ReplayStrategyBuilder}.
+     * @return a {@link ReplayStrategyBuilder} using the historyHint and TTL strategy.
+     */
+    public static <T> ReplayStrategyBuilder<T> historyTtlBuilder(int historyHint, Duration ttl, Executor executor,
```

Comment:
@bryce-anderson - I added this method as a result of your suggestions.

`lazy = false` is likely to be useful if time duration is the dominant limiter, `lazy = true` if history is the dominant limiter.

wdyt?

Reply:
I like it. Thank you.